http.go

```diff
@@ -59,7 +59,12 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	go func() {
 		<-r.Context().Done()
+
 		sub.close()
+
+		if s.AutoStream && len(stream.subscribers) == 0 {
```

Comment:
We should add another option to automatically remove streams when there are no subscribers. 

This means it can be used when not using `AutoStream`. 

Also we may not want it to remove streams when we only have one client that may need to connect back in and collect the messages from a stream.

Reply:
Accessing `len(stream.subscribers) == 0` from the handler will not be thread safe and will cause a data race.

Reply:
> We should add another option to automatically remove streams when there are no subscribers. This means it can be used when not using AutoStream.

I think this is wrong because If I'm not using `AutoStream` it means I created manually the streams and I don't want they die. IMO.

> Also we may not want it to remove streams when we only have one client that may need to connect back in and collect the messages from a stream.

I think you're right here. But maybe only when using `AutoReplay`?

> Accessing `len(stream.subscribers) == 0` from the handler will not be thread safe and will cause a data race.

Here I'm not so good and I need help understand it better.

Should we create a method `.getSubscribersLength()` with `mutex`?

Reply:
> > We should add another option to automatically remove streams when there are no subscribers. This means it can be used when not using AutoStream.
> 
> I think this is wrong because If I'm not using `AutoStream` it means I created manually the streams and I don't want they die. IMO.
> 

Sure, i think that's fair

> > Also we may not want it to remove streams when we only have one client that may need to connect back in and collect the messages from a stream.
> 
> I think you're right here. But maybe only when using `AutoReplay`?
> 

Probably best to handle this by auto-removing the stream if `AutoReplay` is not enabled.

> > Accessing `len(stream.subscribers) == 0` from the handler will not be thread safe and will cause a data race.
> 
> Here I'm not so good and I need help understand it better.
> 
> Should we create a method `.getSubscribersLength()` with `mutex`?

I think a mutex is probably a bit heavy here, we can use atomic integers to track subscriber count. 

If you rebase against master, I have added a safer `getSubscriberCount()` to the stream in #114